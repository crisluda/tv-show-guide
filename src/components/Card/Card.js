import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ show }) => (
  <div className="col s12 m4">
    <div className="card">
      <div className="">
        <img
          style={{ margin: '0 120px' }}
          src={
            show.image && show.image.medium
              ? show.image.medium
              : 'http://static.tvmaze.com/uploads/images/medium_portrait/203/509996.jpg'
          }
          alt=""
        />
      </div>
      <div className="card-content">
        <span
          className="card-title"
          style={{
            fontFamily: 'Dancing Script, cursive',
            color: '#000',
            textAlign: 'center',
          }}
        >
          {show.name}
        </span>
        <p>
          <span
            style={{
              fontFamily: 'Questrial, sans-serif',
              marginRight: '20px',
              color: '#000',
            }}
          >
            TYPE:
          </span>
          {show.type}
        </p>
        <p>
          <span
            style={{
              fontFamily: 'Questrial, sans-serif',
              marginRight: '20px',
              color: '#000',
            }}
          >
            STATUS:
          </span>
          {show.status}
        </p>
        <p>
          <span
            style={{
              fontFamily: 'Questrial, sans-serif',
              marginRight: '20px',
              color: '#000',
            }}
          >
            PREMIERED:
          </span>
          {show.premiered}
        </p>
        <p>
          <span
            style={{
              fontFamily: 'Questrial, sans-serif',
              marginRight: '20px',
              color: '#000',
            }}
          >
            RUNTIME:
          </span>
          {show.runtime}
        </p>
        <p>
          <span
            style={{
              fontFamily: 'Questrial, sans-serif',
              marginRight: '20px',
              color: '#000',
            }}
          >
            LANGUAGE:
          </span>

          {show.language}
        </p>
      </div>
    </div>
  </div>
);

Card.propTypes = {
  show: PropTypes.shape({
    name: PropTypes.string,
    runtime: PropTypes.number,
    premiered: PropTypes.string,
    status: PropTypes.string,
    type: PropTypes.string,
    language: PropTypes.string,
    image: PropTypes.shape({ medium: PropTypes.string }),
  }).isRequired,
};

export default Card;
