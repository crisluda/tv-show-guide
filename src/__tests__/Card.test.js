import React from 'react';
import { render } from '@testing-library/react';
import Card from '../components/Card/Card';

it('renders the Card component correctly', () => {
  const show = {
    name: 'test',
    rating: { average: '6' },
  };
  const { asFragment } = render(<Card show={show} />);
  expect(asFragment()).toMatchSnapshot();
});
