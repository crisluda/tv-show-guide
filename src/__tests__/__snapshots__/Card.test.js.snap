// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card component correctly 1`] = `
<DocumentFragment>
  <div
    class="col s12 m4"
  >
    <div
      class="card"
    >
      <div
        class="card-image"
      >
        <img
          alt=""
          src="http://static.tvmaze.com/uploads/images/medium_portrait/203/509996.jpg"
        />
      </div>
      <div
        class="card-content"
      >
        <span
          class="card-title"
          style="font-family: Dancing Script, cursive; color: rgb(0, 0, 0);"
        >
          test
        </span>
        <p>
          <span
            style="font-family: Questrial, sans-serif; margin-right: 20px; color: rgb(0, 0, 0);"
          >
            TYPE:
          </span>
        </p>
        <p>
          <span
            style="font-family: Questrial, sans-serif; margin-right: 20px; color: rgb(0, 0, 0);"
          >
            STATUS:
          </span>
        </p>
        <p>
          <span
            style="font-family: Questrial, sans-serif; margin-right: 20px; color: rgb(0, 0, 0);"
          >
            PREMIERED:
          </span>
        </p>
        <p>
          <span
            style="font-family: Questrial, sans-serif; margin-right: 20px; color: rgb(0, 0, 0);"
          >
            RUNTIME:
          </span>
        </p>
        <p>
          <span
            style="font-family: Questrial, sans-serif; margin-right: 20px; color: rgb(0, 0, 0);"
          >
            LANGUAGE:
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
